### PR TITLE
Moving average

### PR DESCRIPTION
**Pull request by aminmarashi**

### CODE REVIEW

**Comment by aminmarashi** [comment]
@borisyankov Could you please review the code?

**Comment by aminmarashi** [comment]
@borisyankov Now could you please review?

File: src/exponentialMovingAverage.js [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
-const EMA = (previousDay: number): number => 1;
+const ema = (vals: [], periods: number) => {
+    if (vals.length === 1) {
+      return vals[0];
+    }
 
-export default (data: Candle[], config: ExponentialMovingAverageConfig) => {
-    const { periods } = config;
-    const sma = simpleMovingAverage(data, config);
+    const prev = ema(vals.slice(0, vals.length - 1), periods);
```
Comment (borisyankov) [inline src/exponentialMovingAverage.js]: We'll need to consider perf implications later.
For now, I have also optimized for clarity of code.

**Comment by borisyankov** [comment]
Looks good!

File: src/exponentialMovingAverage.js (hunk 1 above)
Comment (aminmarashi) [inline src/exponentialMovingAverage.js]: Yeah, we can make it more efficient using dynamic programming.

**Pull request merged**